packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -172,6 +173,12 @@ class CupertinoPageRoute<T> extends PageRoute<T> {
 
   @override
   Color get barrierColor => null;
+  
+  /// Allows you to set opaque to false to prevent route reconstruction. 
+  /// The default behavior is true, and declaring false can have side effects. 
+  /// Use caution and awareness that rebuilding is necessary to properly display the back button.
+  @override
+  final bool opaque;
```

Comment:
It seems strange to me to have a child class of page route change this behavior.

Perhaps we should consider adding a new class for this instead, that has the desired behaviors of MaterialPageRoute but also is transparent?